v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,306 @@
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # OpenSSL CA configuration file
+    [ ca ]
+    default_ca = CA_default
+
+    [ CA_default ]
+    default_days = 3660
+    database = index.txt
+    serial = serial.txt 
+    default_md = sha256 
+    copy_extensions = copy
+
+    # Used to create the CA certificate.
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    x509_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = Cockroach CA
+
+    [ extensions ]
+    keyUsage = critical,digitalSignature,nonRepudiation,keyEncipherment,keyCertSign
+    basicConstraints = critical,CA:true,pathlen:1
+
+    # Common policy for nodes and users.
+    [ signing_policy ]
+    organizationName = supplied
+    commonName = supplied
+
+    # Used to sign node certificates.
+    [ signing_node_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = serverAuth,clientAuth
+
+    # Used to sign client certificates.
+    [ signing_client_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = clientAuth
+    ~~~
+
+    {{site.data.alerts.callout_info}}The <code>keyUsage</code> and <code>extendedkeyUsage</code> parameters are vital for CockroachDB functions. You can modify or omit other parameters as per your preferred OpenSSL configuration, but do not omit the <code>keyUsage</code> and <code>extendedkeyUsage</code> parameters. {{site.data.alerts.end}}
+  
+3. Create the CA key using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ openssl genrsa -out my-safe-directory/ca.key 2048 
+    $ chmod 400 my-safe-directory/ca.key
+    ~~~
+
+4. Create the CA certificate using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create CA certificate.
+    $ openssl req \
+      -new \
+      -x509 \
+      -config ca.cnf \
+      -key my-safe-directory/ca.key \
+      -out certs/ca.crt \
+      -days 3660 \
+      -batch
+
+    # Reset database and index files.
+    $ rm -f index.txt serial.txt
+    $ touch index.txt
+    $ echo '01' > serial.txt
+    ~~~
+
+### Create the certificate and key pairs for nodes
+
+1. Create the node.cnf file for the first node and copy the following configuration into it:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # OpenSSL node configuration file
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    req_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = node
+
+    [ extensions ]
+    subjectAltName = DNS:<node-hostname>,DNS:<node-domain>,IP:<IP Address>
```

Comment:
We need to call out that these placeholders need to be replace with real values.

Reply:
Done.